ros-rviz-tf.html

```diff
@@ -0,0 +1,195 @@
+      },
+
+      destroy: function() {
+        this._tfs = [];
+        this._topic.unsubscribe();
+      },
+
+      hide: function() {
+        if (!this.viewer)
+          return;
+
+        for (var i in this._tfs) {
+          this._hideTf(this._tfs[i], i);
+        }
+      },
+
+      _hideTf: function(tf, index) {
+        this.viewer.scene.remove(tf.node);
+
+        tf.visible = false;
+        this.notifyPath('_tfs.' + index + '.visible')
+      },
+
+      show: function() {
+        if (!this.viewer)
+          return;
+
+        for (var i in this._tfs) {
+          this._showTf(this._tfs[i], i);
+        }
+      },
+
+      _showTf: function(tf, index) {
+        this.viewer.addObject(tf.node);
+
+        tf.visible = true;
+        this.notifyPath('_tfs.' + index + '.visible')
+      },
+
+      _updateDisplay: function() {
+        if (undefined === this._topic) {
+          this._topic = new ROSLIB.Topic({
+            ros: this.ros,
+            name: '/tf',
+            messageType: 'tf/tfMessage',
+          });
+        }
+
+        var that = this;
+        this._topic.subscribe(function(msg) {
+          that._onTf(that, msg);
+        });
+      },
+
+      _onTf: function(that, msg) {
+
+        for (var t of msg.transforms) {
+          that._addFrame(t.child_frame_id);
+        }
+      },
+
+      _callback: function(that, frame, tf) {
+        // Check if it already exists
+        var tfNode = undefined;
+        for (var t of that._tfs) {
+          if (t.frame === frame) {
+            tfNode = t.node;
+            break;
+          }
+        }
+
+        // If it doesn't, create it
+        if (undefined === tfNode) {
+          var tfObj = new ROS3D.Axes({
+            shaftRadius: 0.05
+          });
+          tfObj.scale.set(0.1, 0.1, 0.1);
+
+          tfNode = new ROS3D.SceneNode({
+            frameID: that.globalOptions.fixedFrame,
+            tfClient: that.tfClient,
+            object: tfObj,
+            pose: {
+              position: tf.translation,
+              orientation: tf.rotation
+            }
+          });
+
+          that.push('_tfs', {
+            frame: frame,
+            node: tfNode,
+            visible: true
+           });
+
+          if (that.isShown) {
+            that.viewer.addObject(tfNode);
+          }
+        }
+        else {
+          tfNode.position.set(
+            tf.translation.x,
+            tf.translation.y,
+            tf.translation.z);
+          tfNode.quaternion.set(
+            tf.rotation.x,
+            tf.rotation.y,
+            tf.rotation.z,
+            tf.rotation.w)
+        }
+      },
+
+      _addFrame: function(frame) {
+        that = this;
+        this.tfClient.subscribe(frame, function(tf) {
+          that._callback(that, frame, tf);
+        });
+      },
+
+      _toggleFrame: function(e) {
+        var frame = e.target.value;
+        var show = e.target.checked;
+
+        for (var i in this._tfs) {
```

Comment:
how about using [filter](https://www.vinta.com.br/blog/2015/javascript-lambda-and-arrow-functions/) on `_tfs`? Perhaps you can also get the index once you have the element?